src/NuGet.Core/NuGet.Protocol/Plugins/RequestHandlers/SymmetricHandshake.cs

```diff
@@ -96,7 +96,7 @@ public void Dispose()
             }
 
             _connection.MessageDispatcher.RequestHandlers.TryRemove(MessageMethod.Handshake);
-
+            _timeoutCancellationTokenSource.Dispose();
```

Comment:
I can see you doing null condition check for dispose other places, but why not here?
_timeoutCancellationTokenSource?.Dispose();

Reply:
I didn't do the `null` check here because the property `_timeoutCancellationTokenSource` is readonly.